components/admin/lessons/AdminLessonExerciseCard.tsx

```diff
@@ -0,0 +1,137 @@
+import { gql, useMutation } from '@apollo/client'
+import { get } from 'lodash'
+import React, { useState } from 'react'
+import { Collapse, Spinner } from 'react-bootstrap'
+import { Exercise, Maybe, User } from '../../../graphql'
+import styles from '../../../scss/adminLessonExerciseCard.module.scss'
+
+type HeaderProps = { user: User }
+
+const Header = ({ user }: HeaderProps) => {
+  return (
+    <div className={styles.card__header}>
+      <p className={styles.card__header__username}>{user.username}</p>
+      <div className={styles.card__header__contact__container}>
+        <div className={styles.card__header__contact}>
+          <span>{user.email}</span>
+        </div>
+        <div className={styles.card__header__contact}>
+          <span>{user.discordUsername}</span>
+        </div>
+      </div>
+    </div>
+  )
+}
+
+type SectionProps = {
+  header: string
+  paragraph: string
+  explanation?: Maybe<string>
+}
+
+const Section = ({ header, paragraph, explanation }: SectionProps) => {
+  const [show, setShow] = useState(false)
+
+  return (
+    <div className={styles.card__section}>
+      <h3 className={styles.card__section__header}>{header}</h3>
+      <p className={styles.card__section__desc}>{paragraph}</p>
+      {explanation && (
+        <>
+          <button
+            className={styles['card__section--explanation']}
+            onClick={() => setShow(!show)}
+          >
+            {show ? 'Hide' : 'Show'} explanation
+          </button>
+          <Collapse in={show}>
+            <div
+              id="explanation-collapse-text"
+              className={styles.card__section__explanation}
+            >
+              {explanation}
+            </div>
+          </Collapse>
+        </>
+      )}
+    </div>
+  )
+}
+
+const DELETE_EXERCISE = gql`
+  mutation deleteExercise($id: Int!) {
+    deleteExercise(id: $id) {
+      id
+    }
+  }
+`
+
+type FooterProps = {
+  exercise: Exercise
+  onRemove?: (id: number) => void
+  onUnflag?: (id: number) => void
+}
+
+const Footer = ({ exercise, onRemove }: FooterProps) => {
+  const [deleteExercise, { loading }] = useMutation(DELETE_EXERCISE)
```

Comment:
Shouldn't there be a `useDeleteExercise` hook you can use from graphql/index file?

Reply:
I thought so too, but it doesn't seem to exist. I'm not certain of the requirements for a resolver to be auto-generated as a hook.

Reply:
From my understanding, its not that codegen gql generates resolvers as a hook, but rather codegen generates us hooks to use in place of `useMutation`. For these hooks, the functionality should be the same as `useMutation`, except you don't have to input the query yourself. 

Example of graphql/index code that inputs query for us(useGetApp):
https://github.com/garageScript/c0d3-app/blob/master/graphql/index.tsx#L2851

I think the problem might be that you haven't generated a `useDeleteExerciseMutation` hook yet. Please look into trying to use a `useDeleteExerciseMutation` hook that codegen gql generates for us.

Reply:
> I think the problem might be that you haven't made codegen generate a useDeleteExerciseMutation hook yet.

I think it's because I didn't add the flagged fields to the `Exercise` type. Therefore, it couldn't generate the hook.